File: Wiki/C/code/C_Programming_A_Modern_Approach/ch06_Loops/exercises/05.c
// Purpose: Out of these 3 loops, find the one which is not equivalent to the other two:{{{
//
//     while (i < 10) {...}
//     for (; i < 10;) {...}
//     do {...} while (i < 10)
//}}}
// Reference: page 121 (paper) / 146 (ebook)

#include <stdio.h>

    int
main(void)
{
    // A: It's the 3rd loop:
    //
    //     do {...} while (i < 10)
    //
    // Because  it  always  executes  its  body  at  least  once,  even  if  its
    // controlling expression (`i < 10`) is initially false.
    // In contrast,  if `i < 10` is  false, the  other loops will  never execute
    // their body (not even once).

    return 0;
}

File: Wiki/lua/code/Programming_in_Lua_4th_edition/ch01_Getting_Started/exercises/05.lua
-- Purpose: What is the value of the expression `type(nil) == nil`?
-- Reference: page 10 (paper) / 24 (ebook)


-- The evaluation is `false`.
-- That's because the output of `type()` is always a string.
-- For the test to pass, the RHS should be quoted:

--                 v   v
print(type(nil) == 'nil')
--     true
